openedx_learning/core/publishing/api.py

```diff
@@ -0,0 +1,192 @@
+"""
+Publishing API (warning: UNSTABLE, in progress API)
+
+Please look at the models.py file for more information about the kinds of data
+are stored in this app.
+"""
+from datetime import datetime, timezone
+from typing import Optional
+
+from django.db.models import F, QuerySet
+from django.db.transaction import atomic
+
+from .models import (
+    Draft,
+    LearningPackage,
+    Published,
+    PublishLog,
+    PublishLogRecord,
+    PublishableEntity,
+    PublishableEntityVersion,
+)
+from .model_mixins import PublishableContentModelRegistry
+
+
+def create_learning_package(
+    key: str, title: str, created: Optional[datetime] = None
+) -> LearningPackage:
+    """
+    Create a new LearningPackage.
+
+    The ``key`` must be unique.
+
+    Errors that can be raised:
+
+    * django.core.exceptions.ValidationError
+    """
+    if not created:
+        created = datetime.now(tz=timezone.utc)
+
+    package = LearningPackage(
+        key=key,
+        title=title,
+        created=created,
+        updated=created,
+    )
+    package.full_clean()
+    package.save()
+
+    return package
+
+
+def create_publishable_entity(learning_package_id, key, created, created_by):
```

Comment:
As a general comment: The class hierarchy in openedx-learning is becoming non-trivial. That's not suprising, since we're tackling a complex problem, but having type annotations on all of these API functions could make it much easier to grok. I know you're still iterating on the APIs, so don't consider this blocking.

Reply:
Agreed, all api.py functions should have annotations.

Reply:
I'll add those annotations when I add the action that actually checks them in CI. 😞 I need to do some care and feeding for the CI in this next week.

Reply:
👍🏻